modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/descriptor/DomainDescriptorResolver.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.api.descriptor;
+
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.DomainDescriptor;
+
+/**
+ * Resolves the {@link DomainDescriptor} described by the {@link BundleDescriptor}, wrapping the logic to obtain it.
+ *
+ * @since 4.5
+ */
+public interface DomainDescriptorResolver {
+
+  /**
+   * @return the default implementation of a {@link DomainDescriptorResolver}.
+   */
+  static DomainDescriptorResolver domainDescriptorResolver() {
```

Comment:
this should be named `defaultDomainDescriptorResolver` or `noDomainDescriptorResolver `